src/test/java/unit/GcsBucketFileReferenced.java

```diff
@@ -0,0 +1,432 @@
+        "add ref output matches bucket file name");
+
+    // check that the bucket is in the list
+    List<UFGcsBucketFile> matchedResourceList = listBucketResourcesFileWithName(name);
+    assertEquals(1, matchedResourceList.size());
+    UFGcsBucketFile matchedResource = matchedResourceList.get(0);
+    assertEquals(name, matchedResource.name, "list output matches name");
+    assertEquals(
+        externalBucket.getName(), matchedResource.bucketName, "list output matches bucket name");
+    assertEquals(
+        externalBucketBlobName, matchedResource.filePath, "List output matches bucket file name");
+
+    // `terra resource describe --name=$name --format=json`
+    UFGcsBucketFile describeResource =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucketFile.class, "resource", "describe", "--name=" + name);
+
+    // check that the name and bucket name match
+    assertEquals(name, describeResource.name, "describe resource output matches name");
+    assertEquals(
+        externalBucket.getName(),
+        describeResource.bucketName,
+        "describe resource output matches bucket name");
+    assertEquals(
+        externalBucketBlobName,
+        describeResource.filePath,
+        "describe resource output matches bucket file name");
+
+    // `terra resource delete --name=$name`
+    TestCommand.runCommandExpectSuccess("resource", "delete", "--name=" + name, "--quiet");
+  }
+
+  @Test
+  @DisplayName("resolve a referenced bucket file")
+  void resolve() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra resource add-ref gcs-bucket-file --name=$name --bucket-name=$bucketName
+    // --file-path=$filePath`
+    String name = "resolve";
+    TestCommand.runAndParseCommandExpectSuccess(
+        UFGcsBucketFile.class,
+        "resource",
+        "add-ref",
+        "gcs-bucket-file",
+        "--name=" + name,
+        "--bucket-name=" + externalBucket.getName(),
+        "--file-path=" + externalBucketBlobName);
+
+    // `terra resource resolve --name=$name --format=json`
+    String resolved =
+        TestCommand.runAndParseCommandExpectSuccess(
+            String.class, "resource", "resolve", "--name=" + name);
+    assertEquals(
+        ExternalGCSBuckets.getGsPath(externalBucket.getName(), Optional.of(externalBucketBlobName)),
+        resolved,
+        "resolve matches bucket file name");
+
+    // `terra resource resolve --name=$name --format=json --exclude-bucket-prefix`
+    String resolveExcludeBucketPrefix =
+        TestCommand.runAndParseCommandExpectSuccess(
+            String.class, "resource", "resolve", "--name=" + name, "--exclude-bucket-prefix");
+    assertEquals(
+        externalBucket.getName() + "/" + externalBucketBlobName,
+        resolveExcludeBucketPrefix,
+        "resolve matches bucket file name");
```

Comment:
nit: suggest making the assertion message different from the previous one so it's easier to see where it failed in test logs (in my experience, the line numbers don't always match up, and certainly won't if there were any changes).